objects/src/accounts/delta/vault.rs

```diff
@@ -37,6 +41,99 @@ impl AccountVaultDelta {
         }
     }
 
+    /// Merges another delta into this one, overwriting any existing values.
+    ///
+    /// Inputs and the result are validated as part of the merge.
+    pub fn merge(self, other: Self) -> Result<Self, AccountDeltaError> {
+        self.validate()?;
+        other.validate()?;
```

Comment:
Is there a reason to validate inputs/output when merging `AccountValutDelta` but not for other deltas (e.g., `AccountStorageDelta`)?

Reply:
I think we may as well. I originally thought we could just leave it to the caller; but the more I implement the less certain I am of correctness - so adding additional checks is probably good.

I think a good goal when this is refactored is to enforce safe-by-construction alongside it.